pwiz_tools/Skyline/Controls/Graphs/SummaryBarGraphPaneBase.cs

```diff
@@ -36,6 +36,41 @@ public abstract class SummaryBarGraphPaneBase : SummaryGraphPane
     {
         public class ToolTipImplementation : ITipProvider
         {
+            public class TargetCurveList : List<CurveItem>
+            {
+                private SummaryBarGraphPaneBase _parent;
+
+                public TargetCurveList(SummaryBarGraphPaneBase parent)
+                {
+                    _parent = parent;
+                }
+                public new void Add(CurveItem curve)
+                {
+                    //all targets must be on the same axis
+                    if (Count > 0)
+                        Assume.AreEqual(base[0].GetYAxis(_parent), curve.GetYAxis(_parent), @"All target curves for a tooltip must be on the same axis.");
+                    base.Add(curve);
+                }
+
+                public CurveItem ClearAndAdd(CurveItem curve)

```

Comment:
Usually, you clear the list at the very start of the method, and then add stuff later on.
For this reason, "Clear()" and "Add()" are usually separate methods that you call at different times.
A method "ClearAndAdd" would only make sense if you really wanted the behavior to be "If there end up being no curves, then display the previous tooltip text".

Reply:
Well, if there are no curves then there is no tooltips shown. The intention was to make the client code a bit more concise, since the typical pattern is to make sure no old curve are stuck in the collection and then add a new one, one right after the other.